allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -270,36 +277,52 @@ def take_step(self,
         # Outline here: first we'll construct the input to the decoder, which is a concatenation of
         # an embedding of the decoder input (the last action taken) and an attention over the
         # question.
-        decoder_hidden, decoder_context = state.hidden_state
+        decoder_hidden, decoder_context, attended_question = state.hidden_state
         action = state.action_history[-1] if state.action_history else self._start_index
         action_input = Variable(decoder_hidden.data.new(1).long().fill_(action))
         embedded_input = self._action_embedder(action_input)
-        # (batch_size, input_sequence_length)
-        input_weights = self._input_attention(decoder_hidden,
-                                              state.encoder_outputs,
-                                              state.encoder_output_mask)
-        # (batch_size, encoder_output_dim)
-        attended_input = util.weighted_sum(state.encoder_outputs, input_weights)
-        # (batch_size, encoder_output_dim + target_embedding_dim)
-        decoder_input = torch.cat((attended_input, embedded_input), -1)
+
+        if attended_question is None:
+            # Unless this is the first timestep, we've already computed an attention over the input
+            # given the current hidden state.  If it's the first timestep, we compute an initial
```

Comment:
This comment could be clearer. I didn't get `attended_question` being `None` implies that this the first time step from this comment. How about replacing the first sentence with "This means this is the first time step."?

Reply:
I'll fix this by adding a method and calling it in `forward`, as mentioned above.